### DISCUSSION TITLE
Replacement for obsolete Non-demodulating Sensor TSMP58000

### DISCUSSION THREAD

**Discussion by SteveMyers75** [bengtmartensson/IrScrutinizer#527] [Q&A] [answered]
I am upgrading my Arduino UNO setup so I can attempt to decode a Philips SWK-8630 WebTV IR Keyboard.
I tried using a "regular" TSOP34438, but the data received is inconsistant and sending back what I read in doesn't work on the device.

Digikey says the suggested **TSMP58000** on harctoolbox reached End of Life in 2022 and is no longer produced or available. 

The suggested replacement is a **TSMP98000** which looks (to me) to be a good choice.
The specs for it say "**AC coupled response from 30 kHz to 60 kHz, all data formats**" and appear to be the same/better than the TSMP58000.

Has anyone used this in place of the TSMP58000?
Here is a link to the Datasheet directly from Vishay  https://www.vishay.com/docs/82908/tsmp98000.pdf

**Comment by SteveMyers75** [comment, marked as answer]
UPDATE:
Just in case anyone is interested in knowing, I went ahead and bought a TSMP9800 Non-Demodulation Sensor and it functions perfectly.

**Comment by bengtmartensson** [comment]
Thank you for your contribution. I still have to update the harctoolbox.org article...

Having said that, I do not know the exact difference between the 58000 and the 98000. The latter is newer, thus probably "better", but in what sense? Probably some class of disturbances are being suppressed more efficiently?

I still have a number of 58000s around. I doubt that it is "obsolete" just because Vishay decided to change its product lineup.

**Reply by RobThree** [reply]
> I still have a number of 58000s around.

I'm sorry to 'disturb' this discussion, but is there any chance I can buy one, or maybe a few, from you as well as one, or a few TSOP34438 if you still have any left? I've looked all over but the big guys like Mouser that have 'em are *REALLY* expensive when you include shipping and with AliExpress they will arrive sometime march. I was hoping to get started with IrScrutinizer a little earlier than that. I live in the Netherlands; shipping from Germany shouldn't be Mouser-like and I'm happy to pay for your efforts.

Again, I apologize for 'misusing' the discussions for this.

**Reply by bengtmartensson** [reply]
Hi Rob,
note that Mouser ships for free as soon as you reach a certain free-shipping threshold, which is not that astronomic, I think 50€ or something like that. It is not that hard to reach, possibly ordering "things you need anyhow" to stuff the order.  (My experience with ordering from Mouser is excellent.)

If you still need my help, drop me an email at the address on my profile page.

**Reply by RobThree** [reply]
> note that Mouser ships for free as soon as you reach a certain free-shipping threshold, which is not that astronomic

It's not astronomic. But even if I buy 5 of each (I don't usually buy 1 _anyway_ even though I only _need_ 1 of each) I still need to "stuff" my order for €43 and there's currently not much, if anything, I need. I'm just a hobbyist, tinkering and I just came across IrScrutinizer and have an Arduino Nano in stock and wanted to build it to learn some IR codes for some remotes I have laying around.

I'm not shy on parts according my [PartDB](https://github.com/Part-DB/Part-DB-server):

<img width="569" alt="Screenshot 2025-01-30 at 22 48 41" src="https://github.com/user-attachments/assets/2cdf8c9d-20c8-4d42-9023-99ae19213f4d" />

I just usually order from AliExpress (I know, but, again, I'm just messing about, not professionally) and have stuff delivered in a week or 2 tops. It's just that these parts are somehow hard to come by and for some reason take much longer to order. And the longer it takes for parts to arrive, the sooner I'll lose interest in a project. So I'm trying to strike while the iron is hot 😅

The big guys (Mouser, Digikey, Farnell, ...) are just not really feasible for people like me if you need small amounts of cheap parts. There are intermediaries like Sinuss but I checked them and they didn't have one of the parts (can't remember which one exactly).

> If you still need my help, drop me an email at the address on my profile page.

I will. Thank you. I hope I'm not being a bother. It's just that _if_ you have some laying around (and you're willing to part them), stuffing one or maybe a few you can miss in an envelope and sending it my way could get me the parts much quicker and without having to spend over €40 on stuff I don't need 😉

**Comment by SteveMyers75** [comment]
You are correct, the performance specs are almost identical.
The only real difference I see is a slightly lower power Current/Voltage draw, so I think maybe it would run longer on batteries.

I put them side-by-side to compare.

![image](https://github.com/bengtmartensson/IrScrutinizer/assets/13516384/cab4a316-778b-4505-8570-1a778ce1222f)

**Comment by bengtmartensson** [comment]
I have finally updated the Mouser basket. I have not only replaced TSMP58000 by TSMP98000, but also changed TSOP34438 to [TSOP13338](https://www.vishay.com/docs/82805/tsop131.pdf), since the latter is a more modern design.
